drone/src/agent/executor.rs

```diff
@@ -317,7 +317,25 @@ impl<E: Engine> Executor<E> {
                 }
             }
         }
-
+        if let Ok(EngineBackendStatus::Failed { code }) =
+            self.engine.backend_status(spawn_request).await
+        {
+            if let Err(inject_err) = self
+                .backend_to_monitor
+                .try_get_mut(&spawn_request.backend_id)
+                .unwrap()
```

Comment:
I know this `unwrap()` should always succeed but given that this function is somewhat complex I'm a bit uneasy about relying on manual analysis to convince myself of that. Let's at least use `expect()` here rather than unwrap with a message like `Expected backend monitor missing when logging exit code` so that we can grep it if the impossible ever does happen.

Reply:
addressed.